### DISCUSSION TITLE
Is `Machine Deletion Phase Hooks` still recommanded?

### DISCUSSION THREAD

**Discussion by mogliang** [kubernetes-sigs/cluster-api#10390] [Q&A] [answered]
Hey all,

Just found there isn't public doc for this feature `Machine Deletion Phase Hooks`, only a proposal exists https://github.com/kubernetes-sigs/cluster-api/blob/b7b53ca55c82d8ec39c552a24d6c308076c5bd24/docs/proposals/20200602-machine-deletion-phase-hooks.md

Not sure if this feature is still recommanded to use nowadays?

background
===
we are writing k3s controlplane provider, we need remove etcd member AFTER node drain operation. So want to utilize `Machine Deletion Phase Hooks` to remove etcd.
![image](https://github.com/kubernetes-sigs/cluster-api/assets/1578667/c85b910a-3c26-47f7-8ffa-11331e332f4e)

**Comment by sbueringer** [comment, marked as answer]
The current supported way for pre-drain and pre-terminate hooks is to use these annotations: https://github.com/kubernetes-sigs/cluster-api/blob/e0c9cdb94eacd1d91cc24afddc9bd15b27b795fd/api/v1beta1/machine_types.go#L54-L64
